.github/workflows/example-engine-network-permissions.lock.yml

```diff
@@ -0,0 +1,557 @@
+# This file was automatically generated by gh-aw. DO NOT EDIT.
+# To update this file, edit the corresponding .md file and run:
+#   gh aw compile
+
+name: "Secure Web Research Task"
+on:
+  pull_request:
+    branches:
+    - main
+  workflow_dispatch: null
+
+permissions: {}
+
+concurrency:
+  group: "gh-aw-${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}"
+  cancel-in-progress: true
+
+run-name: "Secure Web Research Task"
+
+jobs:
+  secure-web-research-task:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+    outputs:
+      output: ${{ steps.collect_output.outputs.output }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v5
+      - name: Generate Network Permissions Hook
```

Comment:
Make sure this is only generated if the feature is used (I haven't checked the code either way)

Reply:
yes ofc